File: 476-number-complement/number-complement.cpp
class Solution {
public:
    int findComplement(int num) {
        
        string str=bitset<32> (num).to_string();
        auto it=str.find('1');
        string str2;
        str2=str.substr(it);
        for(int i=0;i<str2.size();i++){
            if(str2[i]=='1')str2[i]='0';
            else str2[i]='01';
        }
      
        return stoi(str2,nullptr,2);
    }
};

File: 476-number-complement/number-complement.js
/**
 * @param {number} num
 * @return {number}
 */
var findComplement = function(num) {
    let s=num.toString(2);
    let str="";
    //console.log(s);
    for(let char of s){
    str+=char==='1'?'0':'1';
    }
   // console.log(str)
    return parseInt(str,2);
};